enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,825 @@
+- Tests have been run to ensure the operator does not break the existing payload jobs
+- The operator can report its status via conditions on the Machine API cluster operator
+
+#### Tech Preview -> GA
+
+Once the operator has been released under the `TechPreviewNoUpgrade` feature set, we will continue to enhance the operator
+before marking it as GA:
+- Full E2E testing of the various behaviours described regarding Machine synchronisation
+- E2E testing of MachineSet and ControlPlaneMachineSet conversion
+- E2E testing of MachineHealthCheck workflows in mixed instance management clusters
+- At least one platform has sufficient conversion logic to be useful and allow users on the platform to start migration
+
+Note, we will not require 100% feature parity for promotion to GA.
+Some features supported by Machine API may be supported in conversion only in subsequent releases.
+
+#### Removing a deprecated feature
+
+No features will be deprecated as a part of this enhancement.
+
+### Upgrade / Downgrade Strategy
+
+#### On Upgrade
+
+When upgrading into a release with the new sync controller,
+new Cluster API resources will be created and existing Machine API resources will have the new API fields added.
+
+There should be no effect on the cluster as each of these operations should be a no-op.
+
+#### On Downgrade
+
+On downgrade, the synchronisation of resources will stop.
+To prevent both Machine API and Cluster API resources being reconciled,
+we will make sure that Cluster API controllers in the 4.N-1 release do not reconcile when a mirror resource exists with the same name.
+Independent of the authoritative API.
+
+This will mean that on downgrade all Machine management reverts to Machine API controllers.
+
+### Version Skew Strategy
+
+In 4.N-1, all resources that are mirrored into Cluster API will be managed by the Machine API controllers.
+The mirroring relationship will be based on the names of the resources to create a 1:1 mapping between Machines and MachineSets/ControlPlaneMachineSets in Machine API and Cluster API.
+If the Cluster API controller observes a mirror resource, it will not reconcile, independent of the value of the authoritative annotation.
+
+During upgrade, we expect all Machines to be authoritative in Machine API, however, if resources are marked as authoritative in Cluster API,
+there may be a small amount of time[^10] during upgrade where the resources are not reconciled, between the time that the Machine API controllers are upgraded and the Cluster API controllers are upgraded.
+
+[^10]: We must ensure that Cluster API controllers are upgraded _after_ Machine API controllers to ensure that this upgrade is safe.
+The Machine API controllers must observe the annotation and stop reconciling before the Cluster API controllers observe the annotation and act upon the value.
+This can be done by running the Cluster API resources at run-level 31 for the release in which this upgrade is handled.
```

Comment:
What is run-level 31?

Reply:
It's a CVO thing, already implemented so you don't need to worry about this tbh